src/gui/userstatus.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) by Camila <hello@camila.codes>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef USERSTATUS_H
+#define USERSTATUS_H
+
+#include <QObject>
+#include <QPointer>
+#include <QVariant>
+
+namespace OCC {
+class AccountState;
+class JsonApiJob;
+
+class UserStatus : public QObject
+{
+    Q_OBJECT
+public:
+    explicit UserStatus(AccountState *accountState, QObject *parent = nullptr);
+    QString currentStatus() const;
+    void fetchCurrentStatus();
+
+private Q_SLOTS:
```

Comment:
grepping seems to indicate we're inconsistent but we use the `slots` and `signals` convention more. So I'd say let's not add more cases of `Q_SLOTS` and `Q_SIGNALS`